File: src/test/java/com/gamma/dal/entities/RegionTest.java
package com.gamma.dal.entities;

import com.gamma.dal.util.DatabaseUtil;
import org.hibernate.Session;
import org.junit.Before;
import org.junit.BeforeClass;
import org.junit.Test;

import static org.junit.Assert.*;

/**
 * Created by requinard on 2/21/17.
 */
public class RegionTest {
    @BeforeClass
    public static void setUp(){
    }


    @Test
    public void createRegion() throws Exception {
        Session session = new DatabaseUtil().getSession();
        session.beginTransaction();

        Region region = new Region("MANC UNITED");
        session.save(region);

        session.getTransaction().commit();
        session.close();

        assertNotNull(region.getId());
    }

}

File: src/test/java/com/gamma/dal/entities/RegionTest.kt
package com.gamma.dal.entities

import com.gamma.repository.RegionRepository
import org.junit.Assert.assertNotNull
import org.junit.Test
import org.junit.runner.RunWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.junit4.SpringRunner

/**
 * Created by requinard on 2/21/17.
 */
@RunWith(SpringRunner::class)
@SpringBootTest
class RegionTest {
    @Autowired
    lateinit var regionRepository: RegionRepository

    @Test
    @Throws(Exception::class)
    fun createRegion() {
        val region = Region("MANC UNITED", 1.0, 1.0, 1.0, 1.0)
        regionRepository.save(region)

        assertNotNull(region.uuid)
    }

}
